### ISSUE TITLE
Open Source WinAppDriver

### ISSUE THREAD

**Issue by brett-burkhart** [Microsoft/WinAppDriver#386]
Are there any plans to make WAD open source?  I saw a [reply from 2016](https://github.com/Microsoft/WinAppDriver/issues/84) but wondered if any more thought has been put into this topic.  

With Appium being open source it's nice to see what it's doing when debugging, but once the request gets to WAD then it's a black box.  I've run into a couple scenarios where it would have been great to look at the code to see what it's actually doing rather than guess.  With limited documentation on the Appium Client side it makes it that much harder work through any issues I have.  Just looking at the code is documentation itself.

Also, if it's open source it'll allow the community easily report back any true bugs and improvement suggestions!!

**Comment by DGCBio** [comment]
I'd like to give this one a BIG thumbs up

**Comment by ianceicys** [comment]
Is there any update on the idea of open sourcing win app driver?

**Comment by samirparekh** [comment]
Yes pelase, can we have the source code please. Additionally can we please support windows 7 .

**Comment by yxhaus** [comment]
Another big thumbs up! Please open source WinAppDriver.

**Comment by tautob** [comment]
Thumbs up, please open source WinAppDriver.

**Comment by technoscavenger** [comment]
Support this request too, kindly make WinAppDriver open source.

**Comment by MyAlterEgo123** [comment]
Thumbs up from me also. I have a WPF TextBlock that somehow disappears from the generated XML tree. I have no idea why and I won't know until I get the code.

**Comment by kfrajtak** [comment]
Please try my fully open source alternative https://github.com/kfrajtak/WinAppDriver/. Hope it will help you to detect the problem.

**Comment by MyAlterEgo123** [comment]
kfrajtak - Will it also work for WPF?

**Comment by kfrajtak** [comment]
Not sure :) I have only tested it on a WinForms app. Is is using Windows UI automation API. Give it a try.

**Comment by MyAlterEgo123** [comment]
@kfrajtak - the solution described in https://social.msdn.microsoft.com/Forums/vstudio/en-US/c2deec09-7955-46f5-ba31-61f302c4dbf4/automation-textblocks-not-visible-from-template?forum=wpf worked for me. Seems like the WinAppDriver is relying on WPF peers. 
Thanks!

**Comment by yxhaus** [comment]
Can someone from MS comment on this one please? What are the milestones and targets when this will be made open source (if ever)? 

I am not demanding just wanted to know if this will become open source. I don't want to be in the same boat again like they have abandoned Coded UI. We have written thousands and thousands of assets using Coded UI and now we have to migrate it somewhere.

**Comment by stu4355226** [comment]
Please open source WinAppDriver

**Comment by jpnolin** [comment]
Hi! Any news on open sourcing this thing? 
It seems to be proposed as the default solution to automate Windows application UI. Jumping in this with the current support would place us in a weird spot but its very interesting. Could you give an update?

**Comment by ti82** [comment]
Piling on. This is clearly not being noticed by MS unfortunately, but it really should be an open source framework. The community seems more engaged than does MS. My team is evaluating UI frameworks and the WinAppDriver is much slower than FlaUI, which I suspect is due to the client/server nature of WinAppDriver, but there is no way to tell!
